letsencrypt/models/base_config_settings.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class BaseConfigSettings(models.TransientModel):
+    _inherit = 'base.config.settings'
+
+    letsencrypt_dns_provider = fields.Selection(
+        [('shell', 'Shell')],
+        string='DNS Provider',
+        help='Set this to Shell if you want an external script to be run.')
+    letsencrypt_shell = fields.Text()
+    letsencrypt_url = fields.Char(string='URL')
+
+    def _compute_letsencrypt_needs_dns_provider(self):
+        ir_config_parameter = self.env['ir.config_parameter']
+        altnames = ir_config_parameter.search([
+            ('key', 'like', 'letsencrypt.altname.%')])
+        for rec in self:
+            for name in altnames:
+                if '*' in name:
+                    rec.letsnecrypt_needs_dns_provider = True
+
+    letsencrypt_needs_dns_provider = fields.Boolean(
+        string='Validation needs DNS Provider?',
+        compute='_compute_letsencrypt_needs_dns_provider')
+
+    @api.model
+    def default_get(self, field_list):
+        res = super(BaseConfigSettings, self).default_get(field_list)
+        ir_config_parameter = self.env['ir.config_parameter']
+        res.update({
+            'letsencrypt_dns_provider': ir_config_parameter.get_param(
+                'letsencrypt_dns_provider'),
+            'letsencrypt_url': ir_config_parameter.get_param(
+                'letsencrypt_url'),
+            'letsencrypt_needs_dns_provider': ir_config_parameter.get_param(
```

Comment:
as this is a computed field, no need to read/write it here and below, right?

Reply:
@hbrunn I need to know in my further calls on `letsencrypt.py` this field's value. Reading its value indeed is unnecessary but I think I need to store it somewhere.